SignUp ViewController 작동 확인 요청

MinnieMinwoo opened the pull request:
회원가입 담당하는 뷰컨 만들었는데, 작동 확인 요청드립니다!
테스트 이후 main과 연결은 해제해 둬서 테스트 할 때 main.storyboard에서 storyboard reference로 signup_Viewcontroller 가져온 다음에 버튼하고 segue로 전체화면 연결하면 확인 가능할 거에요

그리고 하면서 추가로 정해야 할거 같다 느낀게
1. imageView, Textfield 네이밍 convention (일단은 IM, TF로 작성했습니다)
2. extension을 따로 file을 만들어서 뺄건지 여부 (회원가입 뷰컨에 맞춰서 사용할 design이여서 그냥 SignUp_ViewController.swift 파일에 집어넣었는데, 의견 있으시면 말씀해 주세요)
이렇게 두가지 있었는데, 이것도 정하면 좋을거 같습니다!

Commit:
feat: Sign Up 화면을 담당하는 ViewController 화면 추가

ViewController 화면 제작, 따로 main ViewController와 연결은 하지 않음
storyboard reference로 segue 연결해서 테스트 가능

Issue Tracker: #001

eomiso commented on frontend-mobile/reddit-clone/reddit-clone/SignUp_ViewController.swift:
```diff
@@ -0,0 +1,67 @@
+        emailTF.addLeftPadding()
+        usernameTF.addLeftPadding()
+        passwordTF.addLeftPadding()
+        }
```
❓
제가 잘 몰라서🥲... 혹시 swift convention에서 closing bracket을 원래 이렇게 처리하나요?

MinnieMinwoo replied:
저부분은 딱히 생각해본적이 없었는데... xcode가 저렇게 잡아줬던거 같아요!

eomiso commented on frontend-mobile/reddit-clone/reddit-clone/SignUp_ViewController.swift:
```diff
@@ -0,0 +1,67 @@
+            gradientLayer.startPoint = CGPoint(x: 0, y: 0)
+            gradientLayer.endPoint = CGPoint(x: 1, y: 0)
+            gradientLayer.frame = self.bounds
+            gradientLayer.cornerRadius = self.frame.height/2
```
👍👍👍👍👍
저 항상 corner radius 정하는 것 때문에 애먹었는데 이렇게 하면 아주 이뻐지는 군요... 감사합니다!

eomiso commented:
방금 확인했습니다!

1. 스토리보드 사용시 협업을 어떻게 할것인가
  이 부분에 대해서 고민했었는데, 뷰별로 스토리보드를 만드는게 가장 적합해 보이는 군요! (저는 코드로 짜고 있었습니다...)

2.

eomiso commented on frontend-mobile/reddit-clone/reddit-clone/SignUp_ViewController.swift:
```diff
@@ -0,0 +1,67 @@
+            self.layer.insertSublayer(gradientLayer, at: 0)
+            self.contentVerticalAlignment = .center
+            self.setTitleColor(UIColor.white, for: .normal)
+            self.titleLabel?.textColor = UIColor.white
```
❗️
<img width="372" alt="image" src="https://user-images.githubusercontent.com/17143087/143495961-5617d17c-a8a2-49df-8879-963afc235b49.png">

여기서 색깔이 잘 변경 되었어야 할 것 같은데 버튼 글씨 색깔이 변하지 않았습니다.

MinnieMinwoo replied:
이부분은 확인해보겠습니다!

eomiso replied:
아 저는 이부분을 attributed string으로 해결했습니다

eomiso commented:
방금 확인했습니다!

#### 1. 스토리보드 사용시 협업을 어떻게 할것인가
저도 부분에 대해서 고민하고 있었는데, 뷰별로 스토리보드를 만드는게 가장 적합해 보이는 군요! (저는 코드로 짜고 있었습니다...)

#### 2. 키보드가 올라올때 continue button과 함께 올라오기 구현은 혹시 어떻게 할지 생각해 보셨나요?
<img width="300" alt="image" src="https://user-images.githubusercontent.com/17143087/143496432-2d20a8e5-3f87-48b4-bb7c-2198df64531d.png">
    저는 일단 이부분 구현해보려고 노력 중인데 혹시 정답을 아신다면 알려주십쇼 ㅠㅜ

#### 3. 키보드가 올라오면서 중앙의 뷰가 scrollview 로 바뀌는 것은 혹시 생각해보셨는지요?
그리고 심지어 자동으로 위치까지 조정이 됩니다ㅠㅜ. 
<img width="414" alt="image" src="https://user-images.githubusercontent.com/17143087/143496860-c60efe67-c665-41d4-bf20-30e4e02e87cb.png">

#### 4. 상단의 cancel, logo, sign up 버튼이 하나의 bar처럼 작동하는데 3번을 구현하기 위해서는 3개가 하나의 view로 묶이는게 좋지 않을까 조심스레 의견을 여쭙니다.

#### 5. ❗️키보드가 안내려 옵니닷!

#### 6. 어쨌든 충분히 머지 가능한 것 같습니다.
❗️표시 된 부분만 해결 되면 바로 머지가능할 것 같습니다!
사실 제가 2번 3번에 꽂혀서 진도가 안나갔는데 민우님께서 해주신 것처럼 디자인적인 부분은 구현할 것들을 취사 선택하고 기능을 빠르게 빠르게 구현하는게 더 애자일스러울 것 같네요ㅠㅜ. 저도 오늘 중으로 머지 리퀘 날리겠습니다.

MinnieMinwoo commented:
1. 제가 코드로 짜는게 익숙하지가 않네요 ㅜㅜ
2. layout 설정하다보니까 저렇게 된거 같은데 이부분은 저도 고민해보겠습니다!
3. 중앙 view는 처음부터 scroll view로 작동되는걸 원했는데 잘 안되네요 구현하려고 하다가 실패한 결과물이 저건데 혹시 좋은 방법 있다면 알려주시면 감사하겠습니다!
4. 조금만 더 자세히 설명해주실 수 있으실까요..? ㅜㅜ
5. 해결하도록 하겠습니다 ㅜㅜ
6. pr 보시느라 수고 많으셨습니다!!!

eomiso commented:
그리고 추가로 네이밍 컨벤션은... 짜다 보니 변수명이 좀 길어지더라도 TextField, View, 등등 풀네임으로 써주는게 오히려 코드 읽기가 편할 것 같습니다...ㅠㅜ 죄송합니다... 원래 민우님이 하시던 방법이  더 바람직한 것 같아요ㅠㅜ

eomiso commented:
> 3. 조금만 더 자세히 설명해주실 수 있으실까요..? ㅜㅜ

그 화면에서 중앙 뷰를 스크롤해서 올려보면 상단이 bar 처럼하나의 뷰에 묶여있더라구요 ㅠㅜ. 혹시 이해안가시면 말씀해주십쇼ㅠㅜ

Commit:
fix: Sign Up 화면을 담당하는 ViewController 화면 추가
#1에서 버튼 color가 흰색으로 변하지 않는 문제 해결
#1에서 키보드가 내려가지 않는 문제 해결 (확인 필요)

Issue Tracker: #1

Commit:
fix: Sign Up 화면을 담당하는 ViewController 화면 추가
#1에서 버튼 color가 흰색으로 변하지 않는 문제 해결
#1에서 키보드가 내려가지 않는 문제 해결 (확인 필요)

Issue Tracker: #1

MinnieMinwoo commented:
뷰 달아주신거 fix하려고 시뮬레이터 말고 아이폰에 앱 build하려 했는데 권한문제가 발생하네요 (issue 확인해 주세요!) ㅜㅜ 그래서 시뮬레이터로 키보드 나오는 코드 넣어서 확인했었는데, 혹시 작동 제대로 되는지 한번 확인 부탁드립니다!

eomiso commented:
아앗 혹시 issue 열어주셨는지요? 이슈에 딱히 올라온게 없어서요...

eomiso commented:
#2 머지리퀘에도 달아두었던 내용인데요 혹시 어떻게 생각하시는지요?
> 그리고 sign-in 이랑 log-in은 거의 하나의 기능으로 묶인다고 봐야할텐데 dev에서 feature/auth-page(가칭) 브랜치를 하나 더 파서 거시서 머지를 해보고 dev에 옮기는 건 어떤가요? 아마 다음 스프린트 때는 이제 TabViewController와 NavigationController 적용하게 될 텐데 저희 두개 브랜치 머지 하느라 Dev 브랜치에서 새로운 브랜치를 못따면 곤란할 것 같아서요!

MinnieMinwoo commented:
코드리뷰하면서 달려고 했는데 말씀하시는대로 하는게 좋을거 같아요!!
